frontend/src/scenes/trends/trendsLogic.ts

```diff
@@ -155,15 +129,26 @@ export const trendsLogic = kea<trendsLogicType<IndexedTrendResult, TrendResponse
         values: [actionsModel, ['actions']],
     },
 
+    actions: () => ({
+        setFilters: (filters, mergeFilters = true) => ({ filters, mergeFilters }),
+        setDisplay: (display) => ({ display }),
+        toggleVisibility: (index: number) => ({ index }),
+        setVisibilityById: (entry: Record<number, boolean>) => ({ entry }),
+        loadMoreBreakdownValues: true,
+        setBreakdownValuesLoading: (loading: boolean) => ({ loading }),
+        toggleLifecycle: (lifecycleName: string) => ({ lifecycleName }),
+        setCachedResults: (filters: Partial<FilterType>, results: TrendResult[]) => ({ filters, results }),
+    }),
+
     loaders: ({ cache, values, props }) => ({
         _results: {
             __default: {} as TrendResponse,
             setCachedResults: ({ results, filters }) => {
-                return { results, filters }
+                return { result: results, filters }
```

Comment:
this seems unclear (as someone fresh to the code!)

`setCachedResults` the action takes results
`_results` the loader renames `results` to `result`
they're used in `loadResults` the loader as `_results.result.map(`

is the renaming to avoid `_results.results`

maybe this links in to the other comments calling out a need for a more general refactor of these related components

Reply:
Yeah, there is a lot of spaghetti code here. This change was a big fix that caused a bug with reloads having any effect earlier, so you can imagine my frustration here :D. I'll make an issue about this big refactoring need the first chance I get.